### DISCUSSION TITLE
The project cannot be started

### DISCUSSION THREAD

**Discussion by Heachy** [polypheny/Polypheny-DB#466] [Q&A] [answered]
### When start Polypheny-DB 
using its main-Class (`dbms/src/main/java/org/polypheny/db/PolyphenyDb.java`)

**I didn't change anything after importing the project, and I got an error when running the startup class.**

**Is it because I haven't configured anything? It doesn’t seem to be written in the official documentation.**

**Hope to get help！ QAQ**

![image](https://github.com/polypheny/Polypheny-DB/assets/90972647/118a1fb0-fc15-49dd-9bdf-1c43ee9fdd90)

![image](https://github.com/polypheny/Polypheny-DB/assets/90972647/c8ff1407-cee5-4305-8a33-1cde2d9c7c43)

**Comment by hennlo** [comment]
Hi @Heachy I see that you are using IntelliJ we have provided a very small [setup guide](https://docs.polypheny.com/en/latest/devs/development-environment#using-intellij) for this.

After having followed these steps you can use our gradle run-dionfigurations within IntelliJ to start Polypheny.
Could you try this.

**Reply by Heachy** [reply]
Yes, I have read the instructions in the document and want to know where **the automatically added run-configurations** in point 5 of the [setup guide](https://docs.polypheny.com/en/latest/devs/development-environment#using-intellij). 

If I want to run the whole program, should I run the main function or the Polypheny-DB configuration in gradle? But no matter which one I run it doesn't work

Sorry, I used maven more before and don't know much about gradle. I'm a little unclear about this, so I'd like to try running the entire program first and then do some research.

Then the running result of the main function is the same as mentioned at the beginning

This is the result of the Polypheny-DB configuration
![image](https://github.com/polypheny/Polypheny-DB/assets/90972647/b75b9a7f-8e1d-4a24-bbb4-b616fcbc7555)

**Reply by hennlo** [reply]
Correct the Polypheny-DB configuration in gradle is the way to go.
But yes indeed seems odd.
Let me try reproducing it and I'll get to you, this is nothing we have observed in that context before.

**Reply by Heachy** [reply]
Thank you for your help！

That means I need to run the Polypheny-DB configuration in gradle first and then run the main function. When this happens now, it’s because there is something in the Polypheny-DB configuration that I haven’t modified, right?

**Reply by hennlo** [reply]
No actually no need to run the `main()` function if you execute the preconfigured gradle config provided by us. The system automatically starts

**Reply by vogti** [reply, marked as answer]
Hi @Heachy The problem is, that you execute the main-Method without building Polypheny-DB first. Polypheny is a modular system consisting of multiple moduls / plugins. The error you get indicates, that the system is unable to find a plugin that provides an implementation for the catalog. To avoid this, either run Polypheny by executing `./gradlew run` or `./gradlew runReset` or use the autoconfigured run configurations in IntelliJ (as mentioned by @hennlo). Please find below a screenshot where to find these run configurations.
![Bildschirmfoto 2024-02-23 um 11 59 48](https://github.com/polypheny/Polypheny-DB/assets/1682115/66678ef9-aab0-4610-a43d-3b6c8c7e21fa)

**Reply by Heachy** [reply]
Thanks for your advice！ @vogti 

But I only have two configurations here.
![image](https://github.com/polypheny/Polypheny-DB/assets/90972647/4caaab4f-4c21-4a81-9e4a-eecae204baee)
Then when I run Polypheny-DB configuration, it displays like this:
![image](https://github.com/polypheny/Polypheny-DB/assets/90972647/b75b9a7f-8e1d-4a24-bbb4-b616fcbc7555)

This is the result after running `./gradlew run`, it has been stuck at 99%
![image](https://github.com/polypheny/Polypheny-DB/assets/90972647/4d4501dd-cb87-4735-8026-8148bb440398)

Or does this mean that the entire project is running?

But when I access `http://localhost:7659/ `it still doesn’t work.

**Reply by Heachy** [reply]
Thank you for the help！@vogti @hennlo 

The project is already running:
![image](https://github.com/polypheny/Polypheny-DB/assets/90972647/97e977b9-a0cc-4edc-819c-1cb2cee10dd0)

My running process is:
1. Run Polypheny-DB in gradle, but an error is reported
2. When running the main function, an error is also reported.
3. I ran `./gradlew run `and it got stuck at 99% and `./gradlew runReset` got stuck at 97%, but then I just ran the main function again and it was fine.

I think it’s quite amazing haha
